woocommerce/woocommerce-ios
# [Mobile Payments] Add known-readers view to the stack, remove old known reader code

**READY FOR REVIEW**

This is the LAST PR in a series of PRs for the "automatically reconnect to known card readers" feature

This PR adds the "known readers" view model and view controller to the stack, removes old known readers logic from the card present payment stores and removes the feature flag this series of PRs was developed under.

The sequence of planned PRs can be found here: #3559 (comment)

To test:
- Ensure lint-ing passes
- Ensure unit tests pass, especially CardReaderSettingsUnknownViewModelTests and CardReaderSettingsKnownViewModelTests

- Complete checkout for at least one non-subscription order using cash on delivery
- Build and run the app
- Proceed to Settings > Manage Card Reader
- Turn on your card reader
- Ensure it does NOT automatically connect to the card reader
- Tap on Connect Card Reader
- When prompted “Do you want to connect to reader CH....” tap “Connect to Reader”
- Stop the app from Xcode, or exit the app by swiping up briefly to see the deck of apps and then flick upwards on the  app

- Leave the reader turned on
- Restart the app by tapping on its icon
- Proceed to Settings > Manage Card Reader
- Turn on your card reader
- Ensure it DOES automatically connect to the card reader
- Stop the app from Xcode, or exit the app by swiping up briefly to see the deck of apps and then flick upwards on the app

- Leave the reader turned on
- Restart the app by tapping on its icon
- Proceed to the order you created above (under the Processing tab)
- Tap on “Collect Payment”
- Ensure it automatically connects (i.e. you are not prompted “Do you want to connect to reader CH....”)
- Complete the payment by tapping the Point of Sale Test Card to the reader
- Tap “No thanks” to decline a receipt
- Stop the app from Xcode, or exit the app by swiping up briefly to see the deck of apps and then flick upwards on the app

- Leave the reader turned on
- Proceed to Settings > Manage Card Reader
- Tap on Disconnect
- Stop the app from Xcode, or exit the app by swiping up briefly to see the deck of apps and then flick upwards on the Woo app

- Leave the reader turned on
- Restart the app by tapping on its icon
- Proceed to Settings > Manage Card Reader
- Ensure it does NOT automatically connect to the card reader
- Tap on Connect Card Reader
- When prompted “Do you want to connect to reader CH....” tap “Connect to Reader”

Note: The "disconnect reader" screen will briefly flash during the collect payment flow. This is a known issue: #4546 

Update release notes:
